rwengine/src/ai/CharacterControllerActivities.hpp

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#ifndef _CONTROLLER_ACTIVITY_HPP_
+#define _CONTROLLER_ACTIVITY_HPP_
+
+#include <ai/CharacterController.hpp>
+
+
+#define DECL_ACTIVITY(act_name)                                               \
+	class act_name : public CharacterController::Activity                 \
+	{                                                                     \
+	public:                                                               \
+		static constexpr auto ActivityName = #act_name;               \
+		std::string name() const override { return ActivityName; };   \
+		bool update(CharacterObject*, CharacterController*) override;
```

Comment:
I already though so.

Well, I don't like this solution either - that's why I asked for feedback specifically for that patch.
Maybe I can come up with something better using templates (or something else - let's see).

So, please do not merge yet. :)